app/controllers/requests_controller.rb

```diff
@@ -199,4 +205,51 @@ def split_requests(requests)
     @pending_requests = requests.select(&:pending?)
     @resolved_requests = requests.reject(&:pending?)
   end
+
+  def enough_resources
+    def get_num_cpu(host)
+      host[:summary].hardware.numCpuCores
+    end
+
+    def get_ram_gb(host)
+      host[:summary].hardware.memorySize.to_i / 1024**2
+    end
+
+    def get_storage_gb(host)
+      host[:summary].host.datastore.sum { |datastore| datastore.summary.freeSpace }.to_i / 1024**2
+    end
+
+    hosts = VmApi.instance.all_hosts
```

Comment:
Using the old `VmApi` here seems to cause timeouts on travis and make the tests slow (and fail without the VPN). Take a look at `app/api/v_sphere` and the classes, as well as the `hosts_controller` or the `vm_controller` and their respective tests to see how the new API is used and can be mocked ;)

Reply:
`VSphere::Host.all` might do the trick in this case :wink: Check out the `hosts_controller` test and the other host tests to see how it's mocked there :D

Reply:
@JoshCoder42 I took the liberty to mock it, so I'm resolving this conversation :v: